AirCasting/SearchAndFollow/CompleteScreen/Contollers/SearchAndFollowCompleteScreenController.swift

```diff
@@ -0,0 +1,65 @@
+// Created by Lunar on 04/05/2022.
+//
+
+import Foundation
+import Resolver
+
+protocol SearchAndFollowCompleteScreenController {
+    func createExternalSession(from session: PartialExternalSession,with downloadedStreamsWithMeasurements: [StreamWithMeasurements]) -> ExternalSessionWithStreamsAndMeasurements
+    func downloadMeasurements(streams: [Int], completion: @escaping (Result<[StreamWithMeasurements], Error>) -> Void)
+}
+
+struct DefaultSearchAndFollowCompleteScreenController: SearchAndFollowCompleteScreenController {
+    @Injected private var service: StreamDownloader
+    
+    func createExternalSession(from session: PartialExternalSession,with downloadedStreamsWithMeasurements: [StreamWithMeasurements]) -> ExternalSessionWithStreamsAndMeasurements {
+        .init(uuid: session.uuid,
+              provider: session.provider,
+              name: session.name,
+              startTime: session.startTime,
+              endTime: session.endTime,
+              longitude: session.longitude,
+              latitude: session.latitude,
+              streams: session.stream.compactMap { stream in
+            
+            guard let downloadedStream = downloadedStreamsWithMeasurements.first(where: { $0.sensorName == stream.sensorName }) else { return nil }
+            
+            let measurements = downloadedStream.measurements
+            return .init(id: stream.id,
+                         unitName: stream.unitName,
+                         unitSymbol: stream.unitSymbol,
+                         measurementShortType: stream.measurementShortType,
+                         measurementType: stream.measurementType,
+                         sensorName: stream.sensorName,
+                         sensorPackageName: stream.sensorPackageName,
+                         thresholdsValues: stream.thresholdsValues,
+                         measurements: measurements.map {.init(value: $0.value, time: DateBuilder.getDateWithTimeIntervalSince1970(Double($0.time/1000)), latitude: $0.latitude, longitude: $0.longitude)})
+        })
+    }
+    
+    func downloadMeasurements(streams: [Int], completion: @escaping (Result<[StreamWithMeasurements], Error>) -> Void) {
+        guard !streams.isEmpty else {
+            completion(.failure(CompletionScreenError.noStreams))
+            return
+        }
+        var results: [Result<StreamWithMeasurements, Error>] = []
```

Comment:
as I already pointed it to Janek - just as a sidenote to raise awareness - this `results` is being accessed from multiple threads for writing. It means we have a data race possibility - probably the most common threading issue. Here it's very unlikely that this will happen because of the fact that it will get wriiten to as a result of networking calls which are very very slow in terms of CPU speeds. But worth noting